lib/schema.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+module.exports = {
+  projects: {
+    url: '/accounts/{accountId}/projects',
+    requiredFields: ['accountId'],
+    list: {
+      type: 'GET'
+    },
+    create: {
+      type: 'POST',
+      requiredFields: ['name', 'sitePrimary']
+    },
+    read: {
+      url: '/{projectId}',
+      type: 'GET',
+      requiredFields: ['projectId']
+    },    
+    update: {
+      url: '/{projectId}',
+      type: 'PATCH',
+      requiredFields: ['projectId']
+    },    
+    delete: {
+      url: '/{projectId}',
+      type: 'DELETE',
+      requiredFields: ['projectId']
```

Comment:
Also for GET, PUT, PATCH, DELETE the id value will always be required. Really this could be as simple as `delete: 'DELETE'` with the path and required field defaulted programmatically.